### PR TITLE
support for ec2 testing

### PR DESCRIPTION
**Pull request by dlukman**
@aboten

### CODE REVIEW

File: .kitchen.ec2.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  instance_type: t2.micro # Free tier instance
+  associate_public_ip: true
+  region: <% @region %>
+  availability_zone: <% @region %>a
+  security_group_ids: [<%= @group_id %>]
+  subnet_id: <%= @subnet_id %>
+
+provisioner:
+  name: chef_zero
+  require_chef_omnibus: true
```
Comment (rveznaver) [inline .kitchen.ec2.yml]: not necessary

File: .kitchen.ec2.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    driver:
+      image_id: ami-1255b321
+    transport:
+      username: centos
+
+suites:
+  - name: default
+    run_list:
+      - recipe[rundeck-node]
+    attributes:
```
Comment (rveznaver) [inline .kitchen.ec2.yml]: also not necessary

File: aws-name-to-id.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Used to find group and subnet ids from their names
+
+require 'aws-sdk'
```
Comment (rveznaver) [inline aws-name-to-id.rb]: error out with an informative message

File: .kitchen.ec2.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# - AWS_SECURITY_GROUP
+# - AWS_SUBNET
+
+<% require_relative 'aws-name-to-id' %>
+
+driver:
+  name: ec2
+  instance_type: t2.micro # Free tier instance
+  associate_public_ip: true
+  region: <% @region %>
```
Comment (rveznaver) [inline .kitchen.ec2.yml]:
``` ruby
region: <%= @region %>
```

File: .kitchen.ec2.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  availability_zone: <% @region %>a
```
Comment (rveznaver) [inline .kitchen.ec2.yml]: Calculate availability zone from received subnet id

File: aws-name-to-id.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+group_name =  ENV['AWS_SECURITY_GROUP'] || 'criteo-sre-core-testing-oss-vpc-security-group'
+subnet_name = ENV['AWS_SUBNET'] || 'criteo-sre-core-testing-oss-vpc-subnet-1'
+@region =     ENV['AWS_REGION'] || 'us-west-2'
+ec2 =         Aws::EC2::Client.new(region: @region)
+
+@group_id = ec2.describe_security_groups(
```
Comment (rveznaver) [inline aws-name-to-id.rb]: clean error message

File: aws-name-to-id.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  filters: [
+    {
+      name:   'group-name',
+      values: [group_name]
+    }
+  ]
+)[0][0].group_id
+
+@subnet_id = ec2.describe_subnets(
```
Comment (rveznaver) [inline aws-name-to-id.rb]: clean error message

File: .kitchen.ec2.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+# - AWS_SUBNET
+
+<% require_relative 'aws-name-to-id' %>
+
+driver:
+  name: ec2
+  instance_type: t2.micro # Free tier instance
+  associate_public_ip: true
+  region: <%= @region %>
+  availability_zone: <%= @region %>
```
Comment (rveznaver) [inline .kitchen.ec2.yml]:
``` ruby
availability_zone: <%= @availability_zone %>
```

**Comment by kamaradclimber** [comment]
good for a first iteration.

I think we must add a ticket into our backlog to have the name->id feature in ~kitchen-ec2 gem

**Comment by kamaradclimber** [comment]
+1 once the typo has been fixed

**Comment by rveznaver** [comment]
According to https://github.com/test-kitchen/kitchen-ec2/issues/83, "if subnet_id is provided the parameters region, availability_zone and security_group_ids" may be calculated.

**Comment by rveznaver** [comment]
Next iteration would be solving this: https://github.com/test-kitchen/kitchen-ec2/issues/83

**Comment by rveznaver** [comment]
+1

**Pull request merged**